react/react.d.ts

```diff
@@ -1843,6 +1843,9 @@ declare namespace __React {
         results?: number;
         security?: string;
         unselectable?: boolean;
+        
+        // Allows aria- and data- Attributes
+        [key: string]: any;
```

Comment:
`: string` is not possible, since typescript understands, that we have an accessor here, which will be able to target any of the defined fields.

Nevertheless the smalles form I can find is `[key: string]: string | string[] | boolean | EventHandler<SyntheticEvent> | {__html: string};` and I will update the PR accordingly.